velox/functions/lib/SubscriptUtil.h

```diff
@@ -167,12 +167,6 @@ class SubscriptImpl : public exec::VectorFunction {
     auto baseArray = decodedArray->base()->as<ArrayVector>();
     auto arrayIndices = decodedArray->indices();
 
-    // Subscript into empty arrays always returns NULLs.
-    if (baseArray->elements()->size() == 0) {
-      return BaseVector::createNullConstant(
-          baseArray->elements()->type(), rows.size(), context.pool());
-    }
-
```

Comment:
By the way, I just thought of an alternative. What if we keep this check here but only check it if `allowOutOfBound` is true? This way we avoid running the `applyToSelectedNoThrow` loop when element vector is empty. I'm not sure how much this would affect the performance though.